packages/openapi3/test/shared-routes.test.ts

```diff
@@ -92,4 +92,30 @@ describe("openapi3: shared routes", () => {
       enum: ["resourceGroup", "subscription"],
     });
   });
+
+  it("emits diagnostic if shared parameters are of different types", async () => {
+    const diagnostics = await diagnoseOpenApiFor(
+      `
+      @service({title: "My Service"})
+      namespace Foo {
+        model Resource {
+          id: string;
+        }
+
+        @route("/sharedroutes/resources", { shared: true })
```

Comment:
Nitpick; not sure if we have to support this...but...

this is valid openapi 2.0:
```json
{
  "swagger": "2.0",
  "info": {
    "title": "Proving a point",
    "version": "1.0"
  },
  "paths": {
    "/sharedroutes/resources": {
      "get": {
        "parameters": [
          {
            "name": "filter",
            "in": "header",
            "type": "string"
          },
          {
            "name": "filter",
            "in": "query",
            "type": "string"
          },
          ],
        "responses": {
          "204": {
            "description": "Ok"
          }
        }
      }
    }
  }
}
```